ludwig/schema/common_fields.py

```diff
@@ -0,0 +1,124 @@
+) -> Field:
+    description = description or (
+        "Whether to add a residual connection to each fully connected layer block. "
+        "Requires all fully connected layers to have the same `output_size`."
+    )
+    parameter_metadata = parameter_metadata or COMMON_METADATA["residual"]
+    return schema_utils.Boolean(
+        default=False,
+        description=description,
+        parameter_metadata=parameter_metadata,
+    )
+
+
+def NumFCLayersField(default: int = 0, description: str = None, parameter_metadata: ParameterMetadata = None) -> Field:
+    description = description or (
+        "Number of stacked fully connected layers to apply. "
+        "Increasing layers adds capacity to the model, enabling it to learn more complex feature interactions."
+    )
+    parameter_metadata = parameter_metadata or COMMON_METADATA["num_fc_layers"]
+    return schema_utils.NonNegativeInteger(
+        default=default,
+        allow_none=False,
+        description=description,
+        parameter_metadata=parameter_metadata,
+    )
+
+
+def NormField(
+    default: Optional[str] = None, description: str = None, parameter_metadata: ParameterMetadata = None
+) -> Field:
+    description = description or "Default normalization applied at the beginnging of fully connected layers."
+    parameter_metadata = parameter_metadata or COMMON_METADATA["norm"]
+    return schema_utils.StringOptions(
+        ["batch", "layer", "ghost"],
+        default=default,
+        allow_none=True,
+        description=description,
+        parameter_metadata=parameter_metadata,
+    )
+
+
+def NormParamsField(description: str = None, parameter_metadata: ParameterMetadata = None) -> Field:
+    description = description or "Default parameters passed to the `norm` module."
+    parameter_metadata = parameter_metadata or COMMON_METADATA["norm_params"]
+    return schema_utils.Dict(
+        description=description,
+        parameter_metadata=parameter_metadata,
+    )
+
+
+def FCLayersField(description: str = None, parameter_metadata: ParameterMetadata = None) -> Field:
+    description = description or (
+        "List of dictionaries containing the parameters of all the fully connected layers. "
+        "The length of the list determines the number of stacked fully connected layers "
+        "and the content of each dictionary determines the parameters for a specific layer. "
+        "The available parameters for each layer are: `activation`, `dropout`, `norm`, `norm_params`, "
+        "`output_size`, `use_bias`, `bias_initializer` and `weights_initializer`. If any of those values "
+        "is missing from the dictionary, the default one provided as a standalone parameter will be used instead."
+    )
+    parameter_metadata = parameter_metadata or COMMON_METADATA["fc_layers"]
+    return schema_utils.DictList(
+        description=description,
+        parameter_metadata=parameter_metadata,
+    )
+
+
+INITIALIZER_SUFFIX = """
```

Comment:
Note that this won't be possible after https://github.com/ludwig-ai/ludwig/pull/3075

Reply:
Yes, true, we will need to refactor this.